src/Microsoft.ApplicationInsights/Managed/Shared/Extensibility/Implementation/Tracing/DiagnosticsTelemetryModule.cs

```diff
@@ -132,12 +183,104 @@ public void Initialize(TelemetryConfiguration configuration)
                             portalSender.Send(traceEvent);
                         }
 
+                        // set up heartbeat
+                        if (this.HeartbeatProvider == null)
+                        {
+                            this.HeartbeatProvider = new HeartbeatProvider();
+                        }
+
+                        this.HeartbeatProvider.Initialize(configuration, this.DiagnosticsInstrumentationKey, this.HeartbeatInterval, this.ExcludedHeartbeatProperties, this.isHeartbeatEnabled);
```

Comment:
this method returns true/false as an indication of an error. Either react on result or remove that logic. Otherwise it's silent fall that impossible to troubleshoot.

Reply:
Yes, that is true - done.